Expunge last two sites of resize_dim

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #32116 Refreshing numel on a stride update is pointless.
* #32114 Delete resize_dim()
* **#32112 Expunge last two sites of resize_dim**

It turns out we already removed these from the CPU version; copy
the changes over.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D19579874](https://our.internmc.facebook.com/intern/diff/D19579874)

ezyang committed:
Expunge last two sites of resize_dim

It turns out we already removed these from the CPU version; copy
the changes over.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit 8f6eaef2:


* **1/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=8f6eaef261351940e4c5a745e547981c85ec277f).

### :detective: 1 new failure recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4221811) pytorch_xla_linux_xenial_py3_6_clang7_build (1/1)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=84347832) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=84347832))
<details>
<summary>
<code>Jan 13 19:46:59 Failed to generate ATEN bindings: ['/var/lib/jenkins/workspace/xla/scripts/generate_code.sh']</code>
</summary>

```
Jan 13 19:46:58 Generated 877 wrappers for /var/lib/jenkins/workspace/xla/scripts/../../torch/csrc/autograd/generated/RegistrationDeclarations.h 
Jan 13 19:46:58 AtenXlaType function missed override: Tensor& bitwise_or_out(Tensor& out, const Tensor& self, const Tensor& other); // bitwise_or_out(Tensor,Tensor,Tensor)->Tensor 
Jan 13 19:46:58 AtenXlaType function missed override: Tensor& bitwise_or_out(Tensor& out, const Tensor& self, Scalar other); // bitwise_or_out(Tensor,Tensor,Scalar)->Tensor 
Jan 13 19:46:58 Traceback (most recent call last): 
Jan 13 19:46:58   File "/var/lib/jenkins/workspace/xla/scripts/gen.py", line 1052, in <module> 
Jan 13 19:46:58     generate(args) 
Jan 13 19:46:58   File "/var/lib/jenkins/workspace/xla/scripts/gen.py", line 1022, in generate 
Jan 13 19:46:58     assert check_overrides(overrides, overridden) 
Jan 13 19:46:58 AssertionError 
Jan 13 19:46:59 Building torch_xla version: 0.8 
Jan 13 19:46:59 Failed to generate ATEN bindings: ['/var/lib/jenkins/workspace/xla/scripts/generate_code.sh'] 
Jan 13 19:46:59 =================== sccache compilation log =================== 
Jan 13 19:46:59 + cleanup 
Jan 13 19:46:59 + retcode=1 
Jan 13 19:46:59 + set +x 
Jan 13 19:46:59 =========== If your build fails, please take a look at the log above for possible reasons =========== 
Jan 13 19:46:59 Compile requests               2525 
Jan 13 19:46:59 Compile requests executed      2268 
Jan 13 19:46:59 Cache hits                     2141 
Jan 13 19:46:59 Cache misses                    115 
Jan 13 19:46:59 Cache timeouts                    0 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

zou3519 commented on aten/src/THC/THCTensor.cpp:
```diff
@@ -185,12 +185,20 @@ void THCTensor_squeeze1d(THCState *state, THCTensor *self, THCTensor *src, int d
 
   if(src->size(dimension) == 1)
   {
+    std::vector<int64_t> newSize(self->dim() - 1);
```
Previously, this function didn't do any dynamic allocations.

Now, there's always a new dynamic allocation. It would be nice to do one of the following:
1) audit the callsites of this function (I don't think there are a lot) and check that those functions aren't important
2) benchmark this change
3) use `SmallVector<int64_t,5>` to avoid performance regressions

ezyang replied:
OK, let's do a SmallVector :)

zou3519 commented on aten/src/THC/THCTensor.cpp:
```diff
@@ -205,17 +213,29 @@ void THCTensor_unsqueeze1d(THCState *state, THCTensor *self, THCTensor *src, int
-  for (d = self->dim()-1; d > dimension; d--) {
-    self->set_size(d, self->size(d-1));
-    self->set_stride(d, self->stride(d-1));
+  std::vector<int64_t> newSize(/* size */ self->dim()+1);
```
Ditto here, we should try to avoid the dynamic allocation or show that it doesn't matter

zou3519 reviewed (approved):
Code looks fine; let's change vector to SmallVector

ezyang committed:
Update on "Expunge last two sites of resize_dim"


It turns out we already removed these from the CPU version; copy
the changes over.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Expunge last two sites of resize_dim"


It turns out we already removed these from the CPU version; copy
the changes over.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Expunge last two sites of resize_dim"


It turns out we already removed these from the CPU version; copy
the changes over.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Expunge last two sites of resize_dim"


It turns out we already removed these from the CPU version; copy
the changes over.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D19579874](https://our.internmc.facebook.com/intern/diff/D19579874)

[ghstack-poisoned]
